### PR TITLE
#5635: SYCL: Add Overloads for parallel_scan with return value for TeamThreadRange

### PR DESCRIPTION
**Pull request by thearusable**
Related to #5635 #6453

~Depends on https://github.com/kokkos/kokkos/pull/6303~ (merged)

### CODE REVIEW

**Comment by masterleinad** [comment]
Marking as draft since dependencies have not been merged.

File: core/src/SYCL/Kokkos_SYCL_Team.hpp [hunk 1]
```diff
@@ -573,28 +573,30 @@ parallel_reduce(const Impl::TeamThreadRangeBoundariesStruct<
...
         loop_bounds,
-    const FunctorType& lambda) {
-  // Extract value_type from lambda
-  using value_type = typename Kokkos::Impl::FunctorAnalysis<
+    const FunctorType& lambda, ValueType& return_val) {
+  // Extract ValueType from the Closure
+  using closure_value_type = typename Kokkos::Impl::FunctorAnalysis<
       Kokkos::Impl::FunctorPatternInterface::SCAN, void, FunctorType,
       void>::value_type;
+  static_assert(std::is_same<closure_value_type, ValueType>::value,
```
Comment (fnrizzi) [inline core/src/SYCL/Kokkos_SYCL_Team.hpp:585]: can you please use _v?

**Review by fnrizzi — APPROVED** [review]
ok with me - caveat my very minor suggestion

File: core/src/SYCL/Kokkos_SYCL_Team.hpp (hunk 1 above)
Comment (cz4rs) [reply core/src/SYCL/Kokkos_SYCL_Team.hpp]: Done.

**Review by masterleinad — APPROVED** [review]

**Comment by cz4rs** [comment]
`OpenMPTarget` failure is unrelated.

**Review by crtrott — APPROVED** [review]

**Pull request merged**